File: src/better_rdp/dpapi.py
"""Thin wrappers over Windows DPAPI (CryptProtectData / CryptUnprotectData).

Windows-only. Heavy imports (win32crypt) are deferred into the function bodies so this
module imports cleanly on any platform for test collection.
"""

from __future__ import annotations


def protect(data: bytes, entropy: bytes = b"") -> bytes:
    """DPAPI-encrypt ``data`` bound to the current Windows user (CurrentUser scope)."""
    raise NotImplementedError


def unprotect(blob: bytes, entropy: bytes = b"") -> bytes:
    """DPAPI-decrypt a blob produced by :func:`protect`."""
    raise NotImplementedError

File: csharp/src/BetterRdp.Core/Dpapi.cs
// Thin wrappers over Windows DPAPI — the C# port of src/better_rdp/dpapi.py.
//
// In .NET this is first-class: System.Security.Cryptography.ProtectedData wraps the same
// Win32 CryptProtectData / CryptUnprotectData the PowerShell smoke (SMOKE.md S1) proved,
// so the blobs are byte-compatible with the Python build.

using System.Security.Cryptography;

namespace BetterRdp.Core;

public static class Dpapi
{
    /// <summary>DPAPI-encrypt <paramref name="data"/> bound to the current Windows user.</summary>
    public static byte[] Protect(byte[] data, byte[]? entropy = null)
        => ProtectedData.Protect(data, entropy, DataProtectionScope.CurrentUser);

    /// <summary>DPAPI-decrypt a blob produced by <see cref="Protect"/>.</summary>
    public static byte[] Unprotect(byte[] blob, byte[]? entropy = null)
        => ProtectedData.Unprotect(blob, entropy, DataProtectionScope.CurrentUser);
}
